src/main/java/ru/test/restservice/service/CompilerService.java

```diff
@@ -0,0 +1,83 @@
+package ru.test.restservice.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static ru.test.restservice.MainApplication.isWindows;
+
+public class CompilerService {
+
+    // TODO: перенести все эксепшны в хандлер
+    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
+    public static class CompilationException extends RuntimeException {
+        public CompilationException(String message) {
+            super(message);
+        }
+    }
+
+    // TODO: добавить функцию сохранения изменений в файле
+
+    /**
+     * Метод, создающий в папке пользователя новый .tex файл
+     * или перезаписывающий уже существующий
+     *
+     * @param text содержимое создаваемого .tex файла
+     */
+    public static void createTexFile(String text) {
+        Path path = Paths.get("test/test.tex");
+        try {
+            if (!Files.exists(Paths.get("test"))) {
+                Files.createDirectory(Paths.get("test"));
+            }
+            if (!Files.exists(path)) {
+                Files.createFile(path);
+            }
+            Files.write(path, text.getBytes(StandardCharsets.UTF_8));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Метод, обращающийся к командной строке для вызова latex компилятора
+     *
+     * @param filename имя компилируемого файла
+     */
+    public static void compileTexFile(String filename) {
+        StringBuilder messageBuilder = new StringBuilder();
+        String line, message = "";
+        try {
+            ProcessBuilder pBuilder = new ProcessBuilder();
+            if (isWindows) {
+                pBuilder.command("cmd.exe", "/c", "pdflatex --synctex=1 --interaction=nonstopmode \"test.tex\"");
+            } else {
+                pBuilder.command("sh", "-c", "pdflatex --synctex=1 --interaction=nonstopmode \"test.tex\"");
```

Comment:
Раз уж добавил параметр, используй
```suggestion
                pBuilder.command("cmd.exe", "/c", "pdflatex --synctex=1 --interaction=nonstopmode \"" + filename + \"");
            } else {
                pBuilder.command("sh", "-c", "pdflatex --synctex=1 --interaction=nonstopmode \"" + filename + "\"");
```